codeworld-server/src/Util.hs

```diff
@@ -36,8 +37,10 @@ import Data.Monoid
 import Data.Text (Text)
 import qualified Data.Text as T
 import qualified Data.Text.Encoding as T
+import qualified Data.Text.Lazy as LT
+import qualified Data.Text.Lazy.Encoding as LT
 import System.Directory
-import System.File.Tree (copyTo_, getDirectory)
+import System.File.Tree (Tree(Node), toTree, copyTo_, getDirectory)
```

Comment:
Here's some bad news.  I'm working on migrating CodeWorld to GHCJS 8.6, and it turns out that I needed to remove uses of `filesystem-trees`.  This package is not maintained, and still depends on `directory` < 1.3, which in turn depends on `base` < 4.11.  That means it cannot be used with modern GHC versions.

We could patch the package locally, but honestly it doesn't do anything really special.  So I'd rather just replace its uses with code that uses `directory` immediately, instead.  I already did this (see the `ghc86` branch) for existing code, but this is adding a new one.

Reply:
Ok, no worries, I'll merge `ghc86` into this branch when work will finished, and will replace it by something more modern.

Reply:
I removed `filesystem-trees` from my logic, it nt needed anymore.